gradle/intTestImage.gradle

```diff
@@ -38,12 +39,46 @@ if (useAllDistribution) {
         from(unpackedPath)
     }
 } else {
+    configurations { selfRuntime }
+    if (project.tasks.findByName("jar")) {
+        dependencies { selfRuntime project }
+    }
+
     intTestImage {
-        with project(":distributions").binDistImage
+        evaluationDependsOn ":smokeTest"
+
+        into('bin') {
+            from { project(':launcher').startScripts.outputs.files }
+            fileMode = 0755
+        }
+
+        def runtimeClasspathConfigurations = (rootProject.configurations.coreRuntime
+                + rootProject.configurations.coreRuntimeExtensions
+                + project.configurations.selfRuntime)
+        if (project.configurations.findByName('integTestRuntime')) {
+            runtimeClasspathConfigurations += project.configurations.integTestRuntimeClasspath
+        }
+        if (project.configurations.findByName('crossVersionTestRuntime')) {
+            runtimeClasspathConfigurations += project.configurations.crossVersionTestRuntimeClasspath
+        }
+        if (project.configurations.findByName('smokeTestRuntime')) {
+            runtimeClasspathConfigurations += project.configurations.smokeTestRuntimeClasspath
+        }
+        if (project.configurations.findByName('performanceTestRuntime')) {
+            runtimeClasspathConfigurations += project.configurations.performanceTestRuntimeClasspath
+        }
+
+        into('lib') {
+            from rootProject.configurations.runtime - (rootProject.configurations.runtime - (runtimeClasspathConfigurations - (rootProject.configurations.gradlePlugins - rootProject.configurations.runtime)))
```

Comment:
My brain is too weak for this expression, is there some simpler way to get what we need? Otherwise, can we introduce some local variables?

Reply:
I actually did use a variable but must have reverted it by mistake. Sorry. 😞 

Similar to the comment above: I could not find a simpler way to figure out what goes into lib and what into lib/plugins - without doing further changes to how this is defined in `build.gradle`.